### ISSUE TITLE
Expose Printer#print_type

### ISSUE THREAD

**Issue by swalkinshaw** [rmosolgo/graphql-ruby#534]
I'm working on documentation for GraphQL schemas and thought it might be useful to have an option to show the definition of a type of IDL. (kind of like a "view source" for normal language API docs).

The easiest way to do this is to call `GraphQL::Schema::Printer#print_type` except it's a private method.

Would you be open to some minor changes to make this a public method? If I manually construct a warden and call the method via `send`, it seems to do exactly what I'm looking for.

I originally thought I could just pass in an `only` lambda with the type name, but it doesn't return the full type definition.

Example:

```ruby
GraphQL::Schema::Printer.print_schema(schema, only: ->(m, ctx) { m.name == 'Product' })
=> "schema {\n}\n\ntype Product {\n\n}"
```

```ruby
GraphQL::Schema::Printer.send(:print_type, warden, type)
=>
# returns full definition
```

**Comment by xuorig** [comment]
maybe add `ObjectType.to_definition` with that :)

**Comment by swalkinshaw** [comment]
That's an even better idea :)

**Comment by rmosolgo** [comment]
:+1: sounds like a great idea! `BaseType#to_definition` seems like a nice API. Optional `only:`/`except:` would be consistent, but feel free to put that off until later.

**Comment by rmosolgo** [comment]
Addressed by #539
